python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -214,6 +215,72 @@ def test_gcs_client_reconnect(ray_start_regular_with_external_redis, auto_reconn
         assert gcs_client.internal_kv_get(b"a", None) == b"b"
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular_with_external_redis",
+    [
+        {
+            **generate_system_config_map(
+                num_heartbeats_timeout=20, gcs_rpc_server_reconnect_timeout_s=3600
```

Comment:
Can we verify these two situations?
1. When GCS went down but the issue is not yet discovered by the periodic check status thread, GCS reconnection still works in this case. (I believe so, via `executor->Retry()`).
2. When GCS went down and there is a background request to GCS (which may be hanging), core worker's main event loop (`ray.get()`?) still works.

Reply:
> When GCS went down but the issue is not yet discovered by the periodic check status thread, GCS reconnection still works in this case. (I believe so, via executor->Retry()).

I updated the code. Here if there is a request going on and it errored because of grpc it'll set the status to down.  So it won't go with Retry, but queued directly.